mesh/src/main/java/science/aist/imaging/service/mesh/triangulation/delaunay/AbstractConstrainedDelaunayTriangulation.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2021 the original author or authors.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package science.aist.imaging.service.mesh.triangulation.delaunay;
+
+import org.jdelaunay.delaunay.ConstrainedMesh;
+import org.jdelaunay.delaunay.error.DelaunayError;
+import org.jdelaunay.delaunay.geometries.DEdge;
+import org.jdelaunay.delaunay.geometries.DPoint;
+import org.jdelaunay.delaunay.geometries.DTriangle;
+import science.aist.imaging.api.domain.AbstractJavaLine;
+import science.aist.imaging.api.domain.AbstractJavaPoint;
+import science.aist.imaging.api.domain.AbstractJavaPolygon;
+import science.aist.imaging.service.mesh.triangulation.Triangulation;
+import science.aist.seshat.Logger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * <p>Abstract implementation of a constrained delaunay triangulation process </p>
+ *
+ * @author Christoph Praschl
+ * @since 1.2
+ */
+public abstract class AbstractConstrainedDelaunayTriangulation<P extends AbstractJavaPoint<P>, L extends AbstractJavaLine<P>, G extends AbstractJavaPolygon<P, L>> implements Triangulation<P, L, G> {
+    private static final Logger logger = Logger.getInstance(AbstractConstrainedDelaunayTriangulation.class);
+    private final Function<P, DPoint> pointConversionFunction = p -> {
+        try {
+            return new DPoint(p.getX(), p.getY(), p.getZ());
+        } catch (DelaunayError delaunayError) {
+            logger.debug("Could not convert given point: " + p);
+            return null;
+        }
+    };
+    private final Function<L, DEdge> lineConversionFunction = l -> new DEdge(pointConversionFunction.apply(l.getStartPoint()), pointConversionFunction.apply(l.getEndPoint()));
+
+    private Function<List<DPoint>, G> polygonFunction;
+    private final Function<DTriangle, G> triangleConversionFunction = t -> polygonFunction.apply(t.getPoints());
+
+
+    /**
+     * Flag that decides if debug level should be used
+     */
+    private final boolean verboseMode;
+
+    /**
+     * <p></p>Force the integrity of the constraints used to compute the delaunay triangulation:</p>
```

Comment:
The p-tag is closing here too soon. This could possibly lead into a javadoc error.